Modified the OAuthHook's __call__ method to treat GET and DELETE methods

ctoth opened the pull request:
Modified the OAuthHook's **call** method to treat GET and DELETE methods the same. Previously the hook completely ignored DELETE requests. This change now works with the Twitter API destroy lists method DELETE /<user>/lists/<id>.

Commit:
Modified the OAuthHook's __call__ method to treat GET and DELETE methods the same. Previously the hook completely ignored DELETE requests. This change now works with the Twitter API destroy lists method DELETE /<user>/lists/<id>.

maraujop commented:
I understand,

Thanks for submitting the patch.

Could you please add a little test. It should be a piece of cake if you look at <a href="https://github.com/maraujop/requests-oauth-hook/blob/dev/tests/tests.py">`tests.py`</a>

If you don't have the time, don't worry, I will do it next week.

Cheers,
Miguel

Commit:
Added a test for DELETE functionality.
This is more of an intigration test than anything as it requires both GET and POST to be working, but it demonstrates that the DELETE method now is signed correctly.

maraujop commented:
Thanks Christopher,

I finally found the time to check it and integrate it. I have bumped the library to version 0.1.2 and uploaded the new version to PyPi. I've also added you to contributors.

Cheers
Miguel
